Mathlib/Analysis/Meromorphic/Order.lean

```diff
@@ -325,9 +363,50 @@ theorem order_ne_top_of_isPreconnected {y : 𝕜} (hU : IsPreconnected U) (h₁x
     (hf y hy).order ≠ ⊤ :=
   (hf.exists_order_ne_top_iff_forall ⟨nonempty_of_mem h₁x, hU⟩).1 (by use ⟨x, h₁x⟩) ⟨y, hy⟩
 
-/-- If the target is a complete space, then the set where a mermorphic function has zero or infinite
-order is discrete within its domain of meromorphicity. -/
-theorem codiscrete_setOf_order_eq_zero_or_top [CompleteSpace E] :
+theorem eventually_analyticAt {f : 𝕜 → E} {x : 𝕜}
+    (h : MeromorphicOn f U) (hx : x ∈ U) : ∀ᶠ y in 𝓝[U \ {x}] x, AnalyticAt 𝕜 f y := by
+  rcases h x hx with ⟨n, g, r, hgr⟩
+  have : U \ {x} ∩ EMetric.ball x r ∈ 𝓝[U \ {x}] x :=
+    inter_mem_nhdsWithin _ (EMetric.ball_mem_nhds _ hgr.r_pos)
+  filter_upwards [this] with y ⟨hy, h'y⟩
+  have hyx : y ≠ x := by simpa using hy.2
+  have : ContinuousAt f y := by
+    have A : ContinuousAt (fun z ↦ (z - x) ^ n • f z) y := by
+      apply hgr.continuousOn.continuousAt
+      exact EMetric.isOpen_ball.mem_nhds h'y
+    have B : ContinuousAt (fun z ↦ (z - x) ^ (-(n : ℤ))) y := by
+      apply ContinuousAt.zpow₀ (by fun_prop)
+      simp [sub_eq_zero, hyx]
+    apply (B.smul A).congr
+    have : {x}ᶜ ∈ 𝓝 y := by simp [hyx]
+    filter_upwards [this] with z hz
+    rw [smul_smul, ← zpow_natCast, ← zpow_add₀]
+    · simp
+    · simpa [sub_eq_zero] using hz
+  exact (h y hy.1).analyticAt this
+
+theorem eventually_analyticAt_or_mem_compl {f : 𝕜 → E} {x : 𝕜}
+    (h : MeromorphicOn f U) (hx : x ∈ U) : ∀ᶠ y in 𝓝[≠] x, AnalyticAt 𝕜 f y ∨ y ∈ Uᶜ := by
+  have : {x}ᶜ = (U \ {x}) ∪ (Uᶜ) := by aesop (add simp Classical.em)
+  rw [this, nhdsWithin_union]
+  simp only [mem_compl_iff, eventually_sup]
+  refine ⟨?_, ?_⟩
+  · filter_upwards [h.eventually_analyticAt hx] with y hy using Or.inl hy
+  · filter_upwards [self_mem_nhdsWithin] with y hy using Or.inr hy
+
+/-- Meromorphic functions on `U` are analytic on `U`, outside of a discrete subset. -/
+theorem analyticAt_mem_codiscreteWithin (hf : MeromorphicOn f U) :
+    { x | AnalyticAt 𝕜 f x } ∈ Filter.codiscreteWithin U := by
+  rw [mem_codiscreteWithin]
+  intro x hx
+  rw [Filter.disjoint_principal_right, ← Filter.eventually_mem_set]
+  filter_upwards [hf.eventually_analyticAt_or_mem_compl hx] with y hy
+  simp
```

Comment:
Non-terminal simp

Reply:
It's followed by `tauto`, which is flexible as far as I can tell.

Reply:
If the flexible linter is happy, so am I!